src/_pytest/fixtures.py

```diff
@@ -304,13 +324,13 @@ def fix_cache_order(
 
 def reorder_items_atscope(
     items: Dict[nodes.Item, None],
-    argkeys_cache: Dict[Scope, Dict[nodes.Item, Dict[_Key, None]]],
-    items_by_argkey: Dict[Scope, Dict[_Key, "Deque[nodes.Item]"]],
+    argkeys_cache: Dict[Scope, Dict[nodes.Item, Dict[FixtureArgKey, None]]],
```

Comment:
@nicoddemus @bluetech i just realized, that this uses the dict of None as set pattern back from the gnarly age of python 2.6 and less

Reply:
Good catch. We could replace that by a set in a follow up.

Reply:
Shall I do this in the PR corresponding to the last item of the list in #11234 ?

Reply:
`dict` is ordered, while `set` is not. The code is very complex and I'm not sure if it matters, but absent a deep analysis I wouldn't change it.